src/robertluo/remote_pull/format.clj

```diff
@@ -45,34 +49,71 @@
       (-> (handler req)
           (assoc :body (-encode formatter ch-out))))))
 
-(defn pull-model
-  [model pattern opt]
-  (let [data (pull/run pattern model)]
-    ((case opt :data-only first :var-only second identity) data)))
+;;; ---------- PULL ----------
 
-(defn with-pull
-  [model-maker pattern-extractor]
+(defn with-pattern
+  [model-maker]
   (fn [req]
-    (if-let [{:keys [pattern opt]} (pattern-extractor req)]
+    (if-let [pattern (-> req :body-params :pattern)]
       (let [model (model-maker req)]
         {:status 200
-         :body   (pull-model model pattern opt)})
+         :body   (pull/run pattern
+                   (if (instance? clojure.lang.Atom model) @model model))})
       (throw (ex-info "No pattern" {:req req})))))
 
-(defn with-pull-sse
-  [model-maker pattern-extractor ch-out]
+(defn with-schema
+  [handler schemas]
   (fn [req]
-    (if-let [{:keys [pattern opt]} (pattern-extractor req)]
-      (let [a-model (model-maker req)
-            data    (pull-model @a-model pattern opt)]
-        (add-watch a-model ::sse (fn [_ _ _ new-v]
-                                   (if new-v
-                                     (let [data (pull-model new-v pattern opt)]
-                                       (async/put! ch-out data))
-                                     (async/close! ch-out))))
-        (async/put! ch-out data)
-        {:status 200})
-      (throw (ex-info "No pattern" {:req req})))))
+    (let [resp (handler req)]
+      (if-let [schema (get schemas (-> req :body-params :schema))]
```

Comment:
`if-let` but just `then` branch, should trigger a warning if you are using clj-kondo.

Reply:
The else branch is present, it's the `resp`.